### ISSUE TITLE
Set bandwidth limit

### ISSUE THREAD

**Issue by 7i77an** [InnovateAsterisk/Browser-Phone#123]
I run 2 demo clients making a video call. 
I have an asterisk 16 that I use as a webrtc server. 
The bandwidth it consumes is 2mb. I tried to modify the SDP with 'b = AS' but it doesn't take the configuration. 
How can I restrict its value?

Thanks.

**Comment by vieridipaola** [comment]
I just wanted to add a word of caution, although I'm aware that what I'm about to state is subjective.

I've been using the VideoHeight setting to limit bandwidth usage just as I've had with other software implementations (eg. Jitsi Meet). In most cases, the video constraints work fine. However, in some specific use cases the video calls fail to establish. For instance, if I explicitly set a video constraint of 320 (tried both Jitsi Meet and Browser-Phone with VideoHeight) some clients will simply "not work" (sorry, I can't provide debugging info just yet). If I unset the constraint, the same failing clients work again.

So there's some black magic going on here that I'm still not fully aware of.

**Comment by InnovateAsterisk** [comment]
Noted, and yes, I see that "video quality" (SD / HD) etc is a separate setting to "Bandwidth", I have noticed that. I think this is something this app should have as a setting.

Btw, the video quality is only going to be what you transmit out (send), you are not in control of what is received. (I'm sure that's obvious, im just making it clear for completeness) 

With regards to the "black magic" I think was able to get to the bottom of this when I was initially doing my tests. The issue is actually to do with the webcam's ability to captures the media stream at the requested size. The GetUserMedia() call comes with a host of different constraints, these can control the width and height and aspect ratios. What I found interesting is even though some webcams can capture at say 720p, when asked to capture at 320, they throw an exception... and the same exception is not consistent from browser to browser. I have about 3/4 different cams and almost all of them behaved differently with constraints, and each reacted differently from one browsers to the next. What i fond is that providing absolutely no media constraint in GetUserMedia() provided the most reliable results by far! But yes... then you are left with a stream bandwidth that isn't controllable.
